File: arrays/jump_game.cs
public class Solution {
    public bool CanJump(int[] nums)
    {
        int lastGood = nums.Length - 1;

        for (int i = nums.Length - 1; i >= 0; i--)
        {
            if (i + nums[i] >= lastGood)
            {
                lastGood = i;
            }
        }

        return lastGood == 0;
    }
}

File: sandbox/jump_game.rb
# @param {Integer[]} nums
# @return {Boolean}
def can_jump(nums)
    n = nums.size
    last_good = n - 1

    (n - 2).downto(0) do |i|
        last_good = i if i + nums[i] >= last_good
    end

    last_good == 0
end
